### ISSUE TITLE
[Android] Include basic proguard file so starting Xamarin Forms project can release with proguard enabled

### ISSUE THREAD

**Issue by PureWeen** [xamarin/Xamarin.Forms#2709]
### Description
The current starter Xamarin Forms Android project will not deploy as is with proguard enabled. 

The minimum configuration currently required is a proguard file with the following entry

```
-keep class android.support.v7.widget.FitWindowsFrameLayout { *; }
-dontwarn android.support.v7.widget.FitWindowsFrameLayout
```

The support libraries all include proguard.txt files as part of their nuget packages so Xamarin Forms could also include the above minimum.

It is debatable whether this is useful since as a user grows an application and uses more things they are going to have to learn how to interact with proguard and modify the file. It is just the inevitable outcome of having proguard enabled. 

Though having the baseline function would probably stop people from creating issues and indicating that it's a defect of Xamarin Forms

### Steps to Reproduce
- Create a new Xamarin Forms Mobile App.
- Under Android options, enabled either "ProGuard" 
- Set Linker to Link SDK Assemblies
- Deploy to a **device.**

### Expected Behavior
App to launch successfully.

### Actual Behavior
Crashes on launch.

### Basic Information

- Version with issue: 3.0.0.446417
- Last known good version: 2.5.1.527436
- IDE: Visual Studio Enterprise 2017 v15.7.1
- Platform Target Frameworks:
  - Android: 8.1

.NET Framework 4.7.2
Android SDK Platform-Tools 27.0.1
Android SDK Build-Tools 27.0.3
Proguard 6.0.3
JDK 8 Update 171
Android NDK R17

### Reproduction Link

[App1.zip](https://github.com/xamarin/Xamarin.Forms/files/1994395/App1.zip)

### RELATED Tickets
https://github.com/xamarin/Xamarin.Forms/issues/2671
https://github.com/xamarin/Xamarin.Forms/issues/2491
https://github.com/xamarin/Xamarin.Forms/issues/2457
https://github.com/xamarin/Xamarin.Forms/issues/2495
#5742

**Comment by jonathanpeppers** [comment]
The way Xamarin.Android works, it creates a rule for any class that generates a Java [ACW](https://developer.xamarin.com/guides/android/advanced_topics/java-integration/android_callable_wrappers/). This is so that any C# type that extends a Java type won't need a proguard rule.

Since the failure here is `android.support.v7.widget.FitWindowsFrameLayout`, Xamarin.Forms is probably not subclassing it, but just using it.

Maybe there is something we need to do in the support library? @Redth ?

**Comment by programmation** [comment]
Having the ProGuard file included in the template is essential IMO. There's already a huge amount of knowledge required to get anything working in the mobile space and it's really important that _all_ the features offered by the system work "out of the box". I agree that every developer that needs ProGuard will need to learn about it, but having the system help us as much as possible by providing expert-level defaults will make a big difference for those just starting out in the learning. 
Anyone who's already an expert already has their own ProGuard file and won't care. But anyone who's learning will really appreciate something to get them started.

**Comment by Redth** [comment]
Yes, I didn't see this rule in particular in the proguard configs for support, so presumably Forms _may_ need to ship their own proguard config file in their nuget package...

**Comment by Redth** [comment]
I think we need to do some more evaluation of other rules that might be needed here.

I also now have an app which needs:

```
-keep class android.support.v7.widget.FitWindowsLinearLayout { *; }
```

**Comment by samhouts** [comment]
closed by #5143
